dir.targets

```diff
@@ -13,25 +13,24 @@
   <Target Name="RebuildAndTest" DependsOnTargets="Rebuild;Test" />
   <Target Name="Test" />
 
-  <Target Name="AnnotateProjectsWithConfiguration"
-          Returns="@(ProjectWithConfiguration)" >
-    <ItemGroup>
-      <ProjectWithConfiguration Include="@(Project)" />
-    </ItemGroup>
-  </Target>
-
+ <!-- build vertical requires the FindBestConfiguration task which is currently in CoreFx.Tools.dll --> 
+ <Import Project="buildvertical.targets" Condition="Exists('$(CoreFxToolsTaskDir)CoreFx.Tools.dll')" /> 
 
-  <!-- build vertical requires the FindBestConfiguration task which is currently in CoreFx.Tools.dll -->
-  <Import Project="buildvertical.targets" Condition="Exists('$(CoreFxToolsTaskDir)CoreFx.Tools.dll')" />
-
-  <Target Name="UndefineTestTFM" BeforeTargets="AssignProjectConfiguration">
+  <Target Name="UndefineTestTFM" 
+          BeforeTargets="AssignProjectConfiguration"
+          Inputs="%(ProjectReference.Identity)"
+          Outputs="fake">
+    <PropertyGroup>
+      <_ProjectReference>%(ProjectReference.Identity)</_ProjectReference>
+      <_UndefineProperties>%(ProjectReference.UndefineProperties)</_UndefineProperties>
+    </PropertyGroup>
     <ItemGroup>
-      <ProjectReference>
         <!--
         Always undefine TestTFM and FilterToTestTFM for all project reference as not needed for compilation and
         avoid bin clash tool to fail
         -->
-        <UndefineProperties>%(ProjectReference.UndefineProperties);TestTFM;FilterToTestTFM</UndefineProperties>
+      <ProjectReference Condition="'%(ProjectReference.Identity)' == '$(ProjectReference)'">
```

Comment:
Where is the property `$(ProjectReference)` defined? I see `$(_ProjectReference)` above. Also what impact does this have on your changes I don't see the connection.